Updated endpoints.

kshakir opened the pull request:
Swagger index.html now re-written (copied/modified from rawls):
- Replaces your-client-id the config google.client_id.
- Scope separator changed from a comma to a space.
- Default yaml changed to cromiam.yaml.
Version and status routes now served without protected /api prefix.
Switched default port for CromIAM to 8001, and the configuration for cromwell back to its default 8000.

geoffjentry commented on src/main/scala/cromiam/webservice/CromIamApiService.scala:
```diff
@@ -54,6 +55,21 @@ trait CromIamApiService extends Directives with SprayJsonSupport with DefaultJso
     Http().singleRequest(cromwellRequest)
   }
 
+  /**
```
Wait, I'm confused about this. We document that version and stats have `api`:

https://github.com/broadinstitute/cromwell#get-apiengineversionversion

geoffjentry replied:
Discussed at standup

geoffjentry commented:
@kshakir What's the plan with this? Fix Cromwell and then un-temporary-fix this? Or did you want this in and worry about cromwell after?

👍 for this either way as long as the near term plan is to just make cromwell right and this to match

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/broadinstitute/CromIAM/pull-request/11/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/CromIAM)

danbills commented on src/main/scala/cromiam/server/config/CromIamServerConfig.scala:
```diff
@@ -10,15 +10,17 @@ import scala.collection.JavaConverters._
 import cromiam.server.config.CromIamServerConfig._
 
-final case class CromIamServerConfig(cromIamConfig: CromIamConfig, cromwellConfig: ServiceConfig, samConfig: ServiceConfig)
+final case class CromIamServerConfig(cromIamConfig: CromIamConfig, cromwellConfig: ServiceConfig,
```
ToL: One param per line is a little easier to read

geoffjentry replied:
for somethingthis short, all on one line would be even easier ;)

kshakir replied:
IntelliJ disagrees on your definition of "short". 😉 

![screen shot 2017-07-17 at 2 16 34 pm](https://user-images.githubusercontent.com/791985/28283142-cecad5a6-6afa-11e7-8cb4-b2834024ae3c.png)

If someone provides a number, _any_ number, I'll create a new config that uses this new margin besides the default. Still hoping we switch to scalafmt at some point, and if you don't give it a number, it picks "80", instead of IntelliJ's liberal "120".

In the meantime, happy to switch to one per line. That's actually what scalafmt would do for this "long" line.

geoffjentry replied:
It's been a long time since I used a VIC-20. 

The reason why I don't like a hard cap is because I disagree with having a hard cap. I like keeping things to around 120-130 but sometimes it's just stupid to force a line wrap at that point (e.g. it's blatantly obvious what comes after anyways) and I haven't seen a formatter yet that can discern the difference.

danbills commented on src/main/scala/cromiam/webservice/CromIamApiService.scala:
```diff
@@ -7,10 +7,12 @@ import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
 import akka.stream.ActorMaterializer
+import cats.instances.future._
+import cats.instances.list._
+import cats.syntax.traverse._
```
I like these imports as much as anybody but are they used?  If not can we put in the compiler flag to enforce this going forward?

kshakir replied:
The compiler flag is [already on](https://github.com/broadinstitute/CromIAM/blob/75f3170db3b8c0483ad1240c3648279791bbb75a/build.sbt#L22-L23).

I just ran `Optimize Imports ⌥⌘O` to canonically tidy up this file. These fancy-pants imports actually come from some scala-fu in https://github.com/broadinstitute/CromIAM/pull/13#discussion_r127582467

geoffjentry replied:
I had a feeling @mcovarr was behind this, somehow.

danbills commented on src/main/scala/cromiam/webservice/SwaggerUiHttpService.scala:
```diff
@@ -54,19 +66,66 @@ trait SwaggerUiHttpService extends Directives {
+  /** Rewrite the swagger index.html. Default passes through the origin data. */
+  protected def rewriteSwaggerIndex(data: String): String = {
+    data
+  }
```
toL no need for braces

danbills commented on src/test/scala/cromiam/webservice/SwaggerUiHttpServiceSpec.scala:
```diff
@@ -39,6 +39,10 @@ object SwaggerUiHttpServiceSpec {
 
+  override protected def rewriteSwaggerIndex(data: String): String = {
+    data.replace("your-client-id", "replaced-client-id")
+  }
```
ToL same as above re: braces

danbills commented:
👍 

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/broadinstitute/CromIAM/pull-request/11/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/CromIAM)

kshakir committed:
Updated endpoints.
Swagger index.html now re-written (copied/modified from rawls):
- Replaces your-client-id the config google.client_id.
- Scope separator changed from a comma to a space.
- Default yaml changed to cromiam.yaml.
Version and status routes now served without protected /api prefix.
Switched default port for CromIAM to 8001, and the configuration for cromwell back to its default 8000.
